backpack/extensions/firstorder/sum_grad_squared/convtranspose1d.py

```diff
@@ -0,0 +1,25 @@
+from torch import einsum, transpose
+
+from backpack.extensions.firstorder.base import FirstOrderModuleExtension
+from backpack.utils import conv_transpose as convUtils
+
+
+class SGSConvTranspose1d(FirstOrderModuleExtension):
+    def __init__(self):
+        super().__init__(params=["bias", "weight"])
+
+    def bias(self, ext, module, g_inp, g_out, backproped):
+        C_axis = 0
+        return (einsum("ncl->nc", g_out[0]) ** 2).sum(C_axis)
+
+    def weight(self, ext, module, g_inp, g_out, backproped):
+        C_axis = 0
+        C_out_axis = 1
+
+        X, dE_dY = convUtils.get_convtranspose1d_weight_gradient_factors(
+            module.input0, g_out[0], module
+        )
+        C_in, C_out, K_X = module.weight.shape
+        d1 = einsum("nml,nkl->nmk", (dE_dY, X))
+        d2 = (d1 ** 2).sum(C_axis).reshape(C_out, C_in, K_X)
+        return transpose(d2, C_axis, C_out_axis)
```

Comment:
same as above. Re axes: We sum out the batch dimension (`N_axis`). Before that, the axis of out channels is `1`. After summing over the batch dimension, the axis of out channels is `0` (because the batch dimension has been summed out). This is currently not correctly represented by the names.

Reply:
Ah right. You are correct that we sum out the batch dimension everywhere (`N_axis`).

So it should be `transpose(d2, C_out_axis, C_in_axis)`, right?
Intuitively atleast, not code wise.